pkg/util/httprequest/httprequest_test.go

```diff
@@ -171,6 +150,21 @@ func TestSchemeHost(t *testing.T) {
 				},
 			},
 			expectedScheme: "https",
+			expectedHost:   "route-namespace.router.default.svc.cluster.local",
+		},
+		"haproxy edge terminated route -> svc -> non-tls pod with the explicit port": {
```

Comment:
same question for this testcase... was this verified with actual backend capture through haproxy?